Src/FluentAssertions/FluentAssertions.csproj

```diff
@@ -44,6 +44,10 @@
   </ItemGroup>
 
   <ItemGroup>
+    <PackageReference Include="Microsoft.Extensions.Configuration" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.EnvironmentVariables" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Json" Version="8.0.0" />
+    <PackageReference Include="Microsoft.Extensions.Configuration.Binder" Version="8.0.0" />
```

Comment:
Wondering if we could put this in a separate package to avoid adding dependencies to the core lib? 
E.g. Everything can be manually configured via code, or one can pull in
`FluentAssertions.Configuration` and do configuration from json or environment variables.

Reply:
Technically, this should be possible, but I seriously doubt that those extra dependencies are going to be an issue. Every modern .NET application will have those packages installed anyway. The only risk is that our version is too high. We could go for version 6.0.0, which is the lowest still supported.

Reply:
While many new applications add those dependencies, FA is also used to test libraries, e.g. other OSS libraries that don't pull in additional dependencies. 

At least I would rather configure FA via code than pulling in extra dependencies.

Reply:
Version 6.x will get our of support in about 11 months. 
I'm here with Jonas: FA (and all packages with high usage):always should have as few as possible dependencies.

Reply:
Hmm, I could ship that as a separate package that uses the new `AssertionEngineInitializerAttribute`, but that won't guarantee it loads at the right time. If another package like `FluentAssertions.DataSets` gets initialized first and updates the equivalency configuration, that new package would overwrite all those changes. Unless, that package would also smart enough to combine the existing configuration with what it retrieved through the `ConfigurationBuilder`.

And what would this package be named? `FluentAssertions.Configuration` implies it's an extension to assert on configurations.

Reply:
In other words, not using `Microsoft.Extensions.Configuration` at all and letting the consumer handle it themselves?

Reply:
Yes. At least in the core library.

I had a look at the xunit and mstest packages and they both have little to no dependencies from other namespaces.